jax/_src/interpreters/pxla.py

```diff
@@ -3259,8 +3259,12 @@ def compile(self,
     return self._executable
 
   def cost_analysis(self) -> Dict[str, float]:
-    return xe.hlo_module_cost_analysis(self.compile_args["backend"],
-                                       self.hlo().as_hlo_module())
+    backend = self.compile_args["backend"]
+    if xb.using_pjrt_c_api(backend):
+      raise NotImplementedError(
+          "Lowered.cost_analysis not implemented on platform "
+          f"'{backend.platform}'. Use compile() to access cost_analysis.")
```

Comment:
When both are implemented, we don't expect `Lowered.cost_analysis` and `Compiled.cost_analysis` to reflect quite the same costs. The former is an estimate prior to compiler optimizations, whereas the latter accounts for them. Maybe tweak to something like this?
```suggestion
          f"'{backend.platform}'. Use compile().cost_analysis() for "
          "post-compilation cost estimates.")
```

Reply:
Done, thanks!